packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { TaskOptions } from '../common/task-protocol';
+import { ILogger, Disposable, DisposableCollection } from '@theia/core/lib/common/';
+import URI from "@theia/core/lib/common/uri";
+import { FileSystemWatcherServer, DidFilesChangedParams, FileChange } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import * as jsoncparser from 'jsonc-parser';
+import { ParseError } from 'jsonc-parser';
+
+export interface TaskConfigurationClient {
+    /**
+     * The task configuration file has changed, so a client might want to refresh its configurations
+     * @returns an array of strings, each one being a task label
+     */
+    taskConfigurationChanged: (event: string[]) => void;
+}
+
+/**
+ * Watches a tasks.json configuration file and provides a parsed version of the contained task configurations
+ */
+@injectable()
+export class TaskConfigurations implements Disposable {
+
+    protected readonly toDispose = new DisposableCollection();
+    protected tasksMap = new Map<string, TaskOptions>();
+    protected watchedConfigFileUri: string;
+
+    /** last directory element under which we look for task config */
+    protected readonly TASKFILEPATH = '.theia';
+    /** task configuration file name */
+    protected readonly TASKFILE = 'tasks.json';
+
+    protected client: TaskConfigurationClient | undefined = undefined;
+
+    constructor(
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(FileSystemWatcherServer) protected readonly watcherServer: FileSystemWatcherServer,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem
+    ) {
+        this.toDispose.push(watcherServer);
+
+        watcherServer.setClient({
```

Comment:
@marcdumais it is bogus, there is only one `FileSystemWatcherServer` which is used in `FileSystemWatcher`. With this change, either `FileSystemWatcher` or `TaskConfiguration` won't receive changes.